etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb

```diff
@@ -0,0 +1,783 @@
+    "\n",
+    "First, let's define a *get_similar* function, essentially a copy from the [scotch_dashboard notebook](./scotch_dashboard.ipynb) to return the data of the top n similar scotches to a given scotch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true,
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def get_similar(name, n, top=True):\n",
+    "    a = sim_df[name].order(ascending=False)\n",
+    "    a.name = 'Similarity'\n",
+    "    df = pd.DataFrame(a) #.join(features_df).iloc[start:end]\n",
+    "    return df.head(n) if top else df.tail(n)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "But since urth-viz-table needs the index (distillery) as a regular column with the data, we need to add that column."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true,
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def get_similar_for_table(name, n=6):\n",
+    "    # select n similar scotches except the first which is the same as name\n",
+    "    simTable = get_similar(name, n).iloc[1:]\n",
+    "    # turn the index (distillery) into a regular column\n",
+    "    return simTable.reset_index()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "We then need another function to retrieve the feature data for a set of scotches as *feature_keys*.\n",
+    "\n",
+    "And because urth-viz-* requires the data series as columns as opposed to rows, we will have to transpose this dataframe.\n",
+    "\n",
+    "Then similarly to the similar table above, we also need to add the index (feature) as a regular column."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true,
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def get_features_for_chart(feature_keys=[]):\n",
+    "    # select the features of the given scotches\n",
+    "    featureTable = features_df.loc[feature_keys]\n",
+    "    # transpose the dataframe and reset the index so that the first column is every feature (smokey, honey, ...)\n",
+    "    # and all the other columns are the series (scotch / distillery names)\n",
+    "    return featureTable.T.reset_index()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "We also need another function that takes a scotch name, a similar scotch name and returns a list of these two names. Scotch is the name user picks from the dropdown, and similar is one user picks from the suggestion table."
```

Comment:
Why is this function being created? How are the widgets going to use it later? (I'm lost on my first read through.)